### ISSUE TITLE
Use pig.js for the gallery/image feed

### ISSUE THREAD

**Issue by Sohalt** [photoprism/photoprism#85]
I think using https://github.com/schlosser/pig.js for the gallery would be really nice. It is similar to Google photos, providing a better user experience than a boring equal sized grid imo.

**Comment by Sohalt** [comment]
Building the Google Photos Web UI – Google Design – Medium - https://medium.com/google-design/google-photos-45b714dfbed1 is a nice read on the Google photos grid.

**Comment by lastzero** [comment]
Thank you for the hint... Looks nice! A contributor could implement that as additional view type 💎

**Comment by rkkoszewski** [comment]
Hi, this sounds interesting, I might help to implement this once I've read trough the Contribution document (Is there any information regarding view types in there?). I actually was about to start a similar open source project like photoprism, but this seems to be a pretty solid project.

**Comment by marcobeierer** [comment]
Hi, I'm also working on a closed-source photo application aimed at professional photographers.

I also had a look at pig.js and the medium article at the time I have implemented the same behavior for my application, but the pig.js demo seemed to be a bit laggy with 4k resolution and the solution in the article seemed unnecessarily complicated.

I just wanna let you know that it is relatively simple to implemented this behavior with pure CSS by using the flexbox technique. The basic concept is described in the following GitHub issue: https://github.com/xieranmaya/blog/issues/6

**Comment by rkkoszewski** [comment]
@marcobeierer I agree with you, but I actually already tested both pig.js (slightly lagy), and also the same flexbox example you sent. Flexbox is way faster, but if you try to show 1000 - 10000 pictures or more at the same time it was even lagging even more (And thats on my desktop PC, I wonder about mobile). But I think a hybrid approach would be best, where flexbox handles the layout but the pictures get loaded/unloaded depending on the view, that should keep flexbox swift, altough if flexbox has too many nodes it also uses cpu quite heavily.

**Comment by danielo515** [comment]
It definitely feels snappier than the current UIs. I think the current UI needs better handling onf lazy loading, but I think that is a topic for a different issue.

**Comment by danielo515** [comment]
> I just wanna let you know that it is relatively simple to implemented this behavior with pure CSS by using the flexbox technique. The basic concept is described in the following GitHub issue: xieranmaya/blog#6

@marcobeierer I don't think the point of pig.js is the layout. It handles lazy loading of images and it's fading out quite well, allows you to scroll the entire page (calculates the total height), and it even handles the cancellation of network request if you scroll so fast that pictures disappear from your screen before they fully load. If you just think about the layout you are probably just seeing the surface

**Comment by SeonD** [comment]
It seems essential to have an API endpoint to return aggregated photo counts by each date (or some other measure) to implement Google Photo kind of mechanism with pig.js. Do we have an endpoint that we can achieve that? If not, does it make sense to include the backend task as a part of this issue?

**Comment by SeonD** [comment]
pig.js works beautifully with PhotoPrism! Check this out! https://drive.google.com/file/d/1Gku6aWRzShX4Ciz3xg68XHIvDCGv_Y8v/view?usp=sharing

**Comment by loulou91** [comment]
This should be a great improvement in the way to present a photos gallery and to be at the level of photos viewers products as GooglePhotos, LibrePhotos...
The major advantage is to be capable to review quickly the complete content of an album as far as no photo is truncated, without opening any single photo.

**Comment by MM-Lehmann** [comment]
Any plans to implement this in the future? Are there ways to push it further up the list somehow?
